core/src/avm2/e4x.rs

```diff
@@ -222,7 +222,15 @@ impl<'gc> E4XNode<'gc> {
                     let is_whitespace_char = |c: &u8| matches!(*c, b'\t' | b'\n' | b'\r' | b' ');
                     let is_whitespace_text = text.iter().all(is_whitespace_char);
                     if !(text.is_empty() || ignore_white && is_whitespace_text) {
-                        let text = AvmString::new_utf8_bytes(activation.context.gc_context, &text);
+                        let text = if matches!(event, Event::Text { .. }) {
+                            AvmString::new_utf8(
+                                activation.context.gc_context,
+                                std::str::from_utf8(&text).unwrap().trim(),
```

Comment:
It'd be nice to avoid `std::str::from_utf8` here and instead trim the byte slice directly (you could write something like [`trim_ascii_start`](https://doc.rust-lang.org/src/core/slice/ascii.rs.html#97), which is currently nightly only but can just be copied here). This would avoid the extra UTF-8 check and potential panic if the text isn't valid UTF-8.